src/urllib3/connection.py

```diff
@@ -213,29 +212,35 @@ def putrequest(self, method, url, *args, **kwargs):
 
         return _HTTPConnection.putrequest(self, method, url, *args, **kwargs)
 
+    def putheader(self, header, *values):
+        """"""
+        if SKIP_HEADER not in values:
+            _HTTPConnection.putheader(self, header, *values)
+        elif six.ensure_str(header.lower()) not in SKIPPABLE_HEADERS:
+            raise ValueError(
+                "urllib3.util.SKIP_HEADER only supports 'Accept-Encoding', 'Host', and 'User-Agent'"
+            )
+
     def request(self, method, url, body=None, headers=None):
-        headers = HTTPHeaderDict(headers if headers is not None else {})
-        if "user-agent" not in headers:
+        if headers is None:
+            headers = {"User-Agent": _get_default_user_agent()}
+        elif "user-agent" not in (k.lower() for k in headers):
             headers["User-Agent"] = _get_default_user_agent()
```

Comment:
I think this flow might change the original `headers`. We even have a test against that in `test_no_user_agent_header` but not when no user agent is given.